Deprecate WKBundlePageFindStringMatches

charliewolfe opened the pull request:
#### f8dbeb24802246ee9b34dfac504c27013a215b0b
<pre>
Deprecate WKBundlePageFindStringMatches
<a href="https://bugs.webkit.org/show_bug.cgi?id=268526">https://bugs.webkit.org/show_bug.cgi?id=268526</a>
<a href="https://rdar.apple.com/122063903">rdar://122063903</a>

Reviewed by Alex Christensen.

Also make changes to make WebKitTestRunner to send a message to the UI process to keep existing tests
working.

Layout tests previously using the injected bundle API need to wait for a presentation update because
it will no longer be synchronous.

* LayoutTests/editing/find/cocoa/find-and-replace-adjacent-words.html:
* LayoutTests/editing/find/cocoa/find-and-replace-at-editing-boundary.html:
* LayoutTests/editing/find/cocoa/find-and-replace-basic.html:
* LayoutTests/editing/find/cocoa/find-and-replace-in-subframes.html:
* LayoutTests/editing/find/cocoa/find-and-replace-noneditable-matches.html:
* LayoutTests/editing/find/cocoa/find-and-replace-replacement-text-input-events.html:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageFindStringMatches):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::findStringMatchesFromInjectedBundle): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::findStringMatchesInPage):
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):

Canonical link: <a href="https://commits.webkit.org/273915@main">https://commits.webkit.org/273915@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/82af78a8fa9fd8fc7b15d49f91987732ffb1b453

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37106 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16011 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39406 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39683 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33139 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38399 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18576 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13149 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31644 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37669 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13504 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32689 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11773 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11775 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33309 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40939 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33600 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34389 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37674 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12090 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9885 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35815 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13731 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/32694 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12462 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4814 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12995 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
```diff
@@ -495,9 +495,8 @@ bool WKBundlePageFindString(WKBundlePageRef pageRef, WKStringRef target, WKFindO
 }
 
-void WKBundlePageFindStringMatches(WKBundlePageRef pageRef, WKStringRef target, WKFindOptions findOptions)
+void WKBundlePageFindStringMatches(WKBundlePageRef, WKStringRef, WKFindOptions)
```
If there were programs that needed to keep finding this in the framework (like SafariForWebKitDevelopment on old OSes) then we would use WK_C_API_DEPRECATED and keep this here.  Since this is only used in WebKitTestRunner and has been for years, let's just remove it.

achristensen07 commented on LayoutTests/editing/find/cocoa/find-and-replace-adjacent-words.html:
```diff
@@ -11,22 +12,30 @@
-testRunner.replaceFindMatchesAtIndices([0, 1, 2], "", false);
-Markup.dump("editor", "After replacing 'apple' with an empty string");
+onload = async () => {
+    testRunner.findStringMatchesInPage("apple", []);
```
Why not make findStringMatchesInPage return a Promise<undefined> and just await it instead of awaiting ensurePresentationUpdate?

charliewolfe replied:
I think I could do this, but the C API would require me to create a page find client to know when the callback should be sent, so this seemed easier.

charliewolfe committed:
Deprecate WKBundlePageFindStringMatches
https://bugs.webkit.org/show_bug.cgi?id=268526
rdar://122063903

Reviewed by Alex Christensen.

Also make changes to make WebKitTestRunner to send a message to the UI process to keep existing tests
working.

Layout tests previously using the injected bundle API need to wait for a presentation update because
it will no longer be synchronous.

* LayoutTests/editing/find/cocoa/find-and-replace-adjacent-words.html:
* LayoutTests/editing/find/cocoa/find-and-replace-at-editing-boundary.html:
* LayoutTests/editing/find/cocoa/find-and-replace-basic.html:
* LayoutTests/editing/find/cocoa/find-and-replace-in-subframes.html:
* LayoutTests/editing/find/cocoa/find-and-replace-noneditable-matches.html:
* LayoutTests/editing/find/cocoa/find-and-replace-replacement-text-input-events.html:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
(WKBundlePageFindStringMatches):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::findStringMatchesFromInjectedBundle): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::findStringMatchesInPage):
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):

Canonical link: https://commits.webkit.org/273915@main
